app/jobs/workflow_execution_status_job.rb

```diff
@@ -30,24 +30,58 @@ def perform(workflow_execution)
     end
 
     wes_connection = Integrations::Ga4ghWesApi::V1::ApiConnection.new.conn
+
+    return if min_run_time && delay_status_check?(wes_connection, workflow_execution, min_run_time)
+
     result = WorkflowExecutions::StatusService.new(workflow_execution, wes_connection).execute
 
     if result
-      queue_next_job(workflow_execution.reload)
+      queue_next_job(workflow_execution.reload, min_run_time)
     else
       handle_unable_to_process_job(workflow_execution, self.class.name)
     end
   end
 
-  def queue_next_job(workflow_execution)
+  # Delay the status check if the workflow is in running state and minimum run time has not been reached
+  def delay_status_check?(wes_connection, workflow_execution, min_run_time)
+    return false if workflow_execution.reload.state == 'running'
+
+    wes_client = Integrations::Ga4ghWesApi::V1::Client.new(conn: wes_connection)
+    status = wes_client.get_run_status(workflow_execution.run_id)
+
+    return false unless status[:state] == 'RUNNING'
+
+    workflow_execution.state = :running
+    workflow_execution.save
+
+    WorkflowExecutionStatusJob.set(wait_until: min_run_time.seconds.from_now).perform_later(workflow_execution,
+                                                                                            min_run_time)
+    true
+  end
+
+  def queue_next_job(workflow_execution, min_run_time = nil)
     case workflow_execution.state.to_sym
     when :canceled, :error
       WorkflowExecutionCleanupJob.perform_later(workflow_execution)
     when :completing
       WorkflowExecutionCompletionJob.perform_later(workflow_execution)
     else
-      WorkflowExecutionStatusJob.set(wait_until: 30.seconds.from_now)
-                                .perform_later(workflow_execution)
+      requeue_or_cancel(workflow_execution, min_run_time)
+    end
+  end
+
+  def requeue_or_cancel(workflow_execution, min_run_time = nil)
+    run_time = state_time_calculation(workflow_execution, :running)
+    maximum_run_time = maximum_run_time(workflow_execution)
+
+    # Max runtime for pipeline (in running state) has exceeded so a cancellation request to WES is made
+    if workflow_execution.cancellable? && maximum_run_time && run_time && (run_time > maximum_run_time)
+      workflow_execution.state = :canceling
```

Comment:
workflow execution states cannot be modified inside of jobs, only inside of services. This is to keep the jobs ACIDic.